Controls/ConnectionControl.cs

```diff
@@ -130,21 +130,40 @@ private void CMB_sysid_SelectedIndexChanged(object sender, EventArgs e)
                     MainV2.comPort.sysidcurrent = temp.sysid;
                     MainV2.comPort.compidcurrent = temp.compid;
 
-                    if (MainV2.comPort.MAV.param.Count == 0 && !(Control.ModifierKeys == Keys.Control))
-                        MainV2.comPort.getParamList();
+                    //force param update and reload view to prevent missing information in config pages
+                    MainV2.comPort.getParamList();
+                    MainV2.View.Reload();

```

Comment:
this change implies that all nodes support the param protocol. which is why it was optional

Reply:
The original code does not account for this either. It only updates if the param list if the current param count is 0. Currently any device that does not support the param protocol will spin forever, trying to fetch parameters unless interrupted by the user

Reply:
also im against a full refresh on changing the target node. if the list has already been received

Reply:
I've dug deeper into this and it looks like there is an issue where if parameters are refreshed on a component that does not have apname set to ARDUPILOTMEGA is causing parameter corruption in mission planner on components with apnames set to ARDUPILOTMEGA. Will have to investigate this further...

Reply:
nice spot on this one. the getInfoFromStream is definitely a bug.
the SetupMavConnect is not technically a bug, as the MAV reference is set inside the same function. but for clarity its better.